src/jit/codegencommon.cpp

```diff
@@ -647,7 +647,7 @@ regMaskTP Compiler::compHelperCallKillSet(CorInfoHelpFunc helper)
 #if defined(_TARGET_AMD64_)
             return RBM_RSI | RBM_RDI | RBM_CALLEE_TRASH;
 #elif defined(_TARGET_ARM64_)
-            return RBM_CALLEE_TRASH_NOGC;
+            return RBM_R12 | RBM_R15;
```

Comment:
The problem I was trying to solve was 
JIT code roughly like this
`
bl JIT_ByRefWriteBarrier
bl JIT_ByRefWriteBarrier
bl JIT_ByRefWriteBarrier
bl JIT_ByRefWriteBarrier
`

Each call is expected to copy one GCRef from *src to *dest and increment both pointers.  The incremented pointers cannot be marked as killed because that causes garbage collector hole.

On the src and dest pointers which will be reused.  It is effectively part of the sematics of the call.